app/models/manageiq/providers/redhat/inventory/collector.rb

```diff
@@ -53,6 +53,24 @@ def collect_datacenters
     end
   end
 
+  def collect_cluster_for_host(host)
+    manager.with_provider_connection(VERSION_HASH) do |connection|
+      connection.follow_link(host.cluster)
+    end
+  end
+
+  def collect_host_stats(host)
+    manager.with_provider_connection(VERSION_HASH) do |connection|
+      connection.link?(host.statistics) ? connection.follow_link(host.statistics) : host.statistics
```

Comment:
It looks like host.statistics can be nil in situation like host being installed so there are no stats yet.